### PR TITLE
[Marketplace Contribution] - EXPANDR - 7685 - Azure Enrichment Playbook Update (Azure Resource Graph)

### PR DESCRIPTION
**Pull request by BigEasyJ**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
[EXPANDR-7685](https://jira-dc.paloaltonetworks.com/browse/EXPANDR-7685)

## Description
Updated the Azure - Enrichment playbook with tasks to support gathering the hierarchy path of an asset from the Azure Resource Graph integration.

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/Azure-Enrichment-Remediation/pack_metadata.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
     "name": "Azure Enrichment and Remediation",
     "description": "Playbooks using multiple Azure content packs for enrichment and remediation purposes",
     "support": "xsoar",
-    "currentVersion": "1.1.17",
+    "currentVersion": "1.1.18",
```
Comment (johnnywilkes) [inline Packs/Azure-Enrichment-Remediation/pack_metadata.json:5]: should we have some other (at least optional) dependencies in here?

File: Packs/Azure-Enrichment-Remediation/Playbooks/Azure_-_Enrichment.yml [hunk 2]
```diff
@@ -2,750 +2,952 @@ id: Azure - Enrichment
...
+- contextPath: MSGraphIdentity.Role
+  description: Microsoft Graph Role object that contains role ID and user IDs.
+  type: unknown
+- contextPath: Account
+  description: Microsoft Graph User account objects for found user IDs.
+  type: unknown
+- contextPath: MSGraphUser
+  description: Microsoft Graph User objects for found user IDs.
+  type: unknown
+- contextPath: azureAssetHierarchy
```
Comment (johnnywilkes) [inline Packs/Azure-Enrichment-Remediation/Playbooks/Azure_-_Enrichment.yml:30]:
```suggestion
- contextPath: AzureAssetHierarchy
```
NIT make this camel case as all others are

File: Packs/Azure-Enrichment-Remediation/Playbooks/Azure_-_Enrichment.yml [hunk 3]
```diff
@@ -2,750 +2,952 @@ id: Azure - Enrichment
...
+    isautoswitchedtoquietmode: false
+    isoversize: false
+    nexttasks:
+      '#none#':
+      - "8"
+    note: false
+    quietmode: 0
+    scriptarguments:
+      key:
+        simple: azureAssetHierarchy
```
Comment (johnnywilkes) [inline Packs/Azure-Enrichment-Remediation/Playbooks/Azure_-_Enrichment.yml:791]:
```suggestion
        simple: AzureAssetHierarchy
```
NIT make this camel case as all others are

File: Packs/Azure-Enrichment-Remediation/Playbooks/Azure_-_Enrichment.yml [hunk 4]
```diff
@@ -2,750 +2,952 @@ id: Azure - Enrichment
...
+      value:
+        complex:
+          accessor: displayName
```
Comment (johnnywilkes) [inline Packs/Azure-Enrichment-Remediation/Playbooks/Azure_-_Enrichment.yml:794]: I could be wrong but from [this](https://prod-xsiam-master-branch1.xdr.us.paloaltonetworks.com/alerts?action:openAlertDetails=196-workPlan-916) test, it seems like the hierarchy base object is the public IP which doesn't make much sense to me

**Review by johnnywilkes — CHANGES_REQUESTED** [review]
@BigEasyJ , check these out, please

File: Packs/Azure-Enrichment-Remediation/Playbooks/Azure_-_Enrichment.yml (hunk 4 above)
Comment (BigEasyJ) [reply Packs/Azure-Enrichment-Remediation/Playbooks/Azure_-_Enrichment.yml:794]: check the `resourceGroup` name after the migration ;) I thought the same thing lol.

Comment (johnnywilkes) [reply Packs/Azure-Enrichment-Remediation/Playbooks/Azure_-_Enrichment.yml:794]: okay, my apologies, should have looked more.  Bad RG name

File: Packs/Azure-Enrichment-Remediation/pack_metadata.json (hunk 1 above)
Comment (BigEasyJ) [reply Packs/Azure-Enrichment-Remediation/pack_metadata.json:5]: Are dependencies a requirement, if so multiple others are missing.

Comment (johnnywilkes) [reply Packs/Azure-Enrichment-Remediation/pack_metadata.json:5]: I think we should list what we have at least as optional

File: Packs/Azure-Enrichment-Remediation/Playbooks/Azure_-_Enrichment_README.md [hunk 5]
```diff
@@ -45,6 +47,7 @@ This playbook does not use any scripts.
 | MSGraphIdentity.Role | Microsoft Graph Role object that contains role ID and user IDs | unknown |
 | Account | Microsoft Graph User account objects for found user IDs. | unknown |
 | MSGraphUser | Microsoft Graph User objects for found user IDs. | unknown |
+| azureAssetHierarchy | Azure Compute Hierarchy Path. | unknown |
```
Comment (johnnywilkes) [inline Packs/Azure-Enrichment-Remediation/Playbooks/Azure_-_Enrichment_README.md:50]:
```suggestion
| AzureAssetHierarchy | Azure Compute Hierarchy Path. | unknown |
```
change/re-run sdk docs

File: Packs/Azure-Enrichment-Remediation/Playbooks/Azure_-_Enrichment.yml [hunk 6]
```diff
@@ -2,750 +2,952 @@ id: Azure - Enrichment
 version: -1
 name: Azure - Enrichment
 description: Given an IP address, this playbook enriches Azure Compute, Microsoft Graph User, and IAM information and outputs Azure Compute, Microsoft Graph User, and IAM information.
+inputs:
+- description: Azure IP in Alert
```
Comment (ShirleyDenkberg) [inline Packs/Azure-Enrichment-Remediation/Playbooks/Azure_-_Enrichment.yml:6]:
```suggestion
- description: Azure IP address in the alert.
```

File: Packs/Azure-Enrichment-Remediation/Playbooks/Azure_-_Enrichment.yml [hunk 7]
```diff
@@ -2,750 +2,952 @@ id: Azure - Enrichment
...
-            value:
-              simple: Azure.Network.IPConfigurations.PublicConfigID
-            iscontext: true
-          right:
-            value: {}
-    continueonerrortype: ""
+    skipunavailable: false
+    task:
+      brand: ""
+      description: Checks whether the last command returned information on the Public IP or not.
```
Comment (ShirleyDenkberg) [inline Packs/Azure-Enrichment-Remediation/Playbooks/Azure_-_Enrichment.yml:194]:
```suggestion
      description: Checks whether the last command returned information on the public IP address or not.
```

File: Packs/Azure-Enrichment-Remediation/Playbooks/Azure_-_Enrichment.yml [hunk 8]
```diff
@@ -2,750 +2,952 @@ id: Azure - Enrichment
...
+      "yes":
+      - "6"
+    note: false
+    quietmode: 0
+    separatecontext: false
+    skipunavailable: false
     task:
-      id: 1f001609-43a6-4284-81d0-10b4089b3598
+      brand: ""
+      description: Check whether the last command returned information for the NIC associated with the IP or not.
```
Comment (ShirleyDenkberg) [inline Packs/Azure-Enrichment-Remediation/Playbooks/Azure_-_Enrichment.yml:296]:
```suggestion
      description: Check whether the last command returned information for the NIC associated with the IP address or not.
```

File: Packs/Azure-Enrichment-Remediation/Playbooks/Azure_-_Enrichment.yml [hunk 9]
```diff
@@ -2,750 +2,952 @@ id: Azure - Enrichment
...
               field:
                 value:
                   simple: id
+            operator: getField
     separatecontext: false
-    continueonerrortype: ""
+    skipunavailable: false
+    task:
+      brand: MicrosoftGraphIdentityandAccess
+      description: Gets all members in the Global admins role ID.
```
Comment (ShirleyDenkberg) [inline Packs/Azure-Enrichment-Remediation/Playbooks/Azure_-_Enrichment.yml:618]:
```suggestion
      description: Gets all members in the Global Admins role ID.
```

File: Packs/Azure-Enrichment-Remediation/README.md [hunk 10]
```diff
@@ -1,19 +1,40 @@
 ##### What does this pack do?
 
-The pack is intended to contain Azure playbooks that conduct enrichment and/or remediation and can use multiple other Azure 
+The pack is intended to contain Azure playbooks and scripts that conduct enrichment and/or remediation and can use multiple other Azure
```
Comment (ShirleyDenkberg) [inline Packs/Azure-Enrichment-Remediation/README.md:3]:
```suggestion
The pack contains Azure playbooks and scripts that conduct enrichment and/or remediation and can use multiple other Azure 
```

File: Packs/Azure-Enrichment-Remediation/README.md [hunk 11]
```diff
@@ -1,19 +1,40 @@
...
-as Compute and MSGraphUsers), they can't reside in one of the current packs because they include content from multiple integrations.  This 
-pack was created as a place to put Azure playbooks that use Azure integrations from multiple packs with a focus on enrichment and 
-remediation.
+as Compute and MSGraphUsers), they can't reside in one of the current packs because they include content from multiple integrations.  This pack was created as a place to put Azure playbooks that use Azure integrations from multiple packs with a focus on enrichment and remediation.
 
+### Scripts
+
+#### AzureFindAvailableNSGPriorities
+
+This script takes in a list of numbers that represent Azure priorities for NSG rules, a target priority number, and a number available priorities to return available priorities from the provided list.
```
Comment (ShirleyDenkberg) [inline Packs/Azure-Enrichment-Remediation/README.md:14]:
```suggestion
This script takes in a list of numbers that represent Azure priorities for NSG rules, a target priority number, and a number of available priorities to return available priorities from the provided list.
```

**Comment by ShirleyDenkberg** [comment]
@johnnywilkes @ssokolovich @thefrieddan1 Doc review completed.

File: Packs/Azure-Enrichment-Remediation/pack_metadata.json (hunk 1 above)
Comment (johnnywilkes) [reply Packs/Azure-Enrichment-Remediation/pack_metadata.json:5]: thanks

**Review by johnnywilkes — APPROVED** [review]
@BigEasyJ , LGTM

**Review by thefrieddan1 — APPROVED** [review]

**Pull request merged**